app/forms/concerns/piv_cac_form_helpers.rb

```diff
@@ -21,6 +21,10 @@ def not_error_token
     possible_error = @data['error']
     if possible_error
       self.error_type = possible_error
+      errors.add(
+        :token, I18n.t('headings.piv_cac.certificate.invalid'),
+        type: possible_error.match(/\.(.*)/)[1].to_sym
```

Comment:
is this similar to `possible_error.split('.').last`? alternatively, we could enumerate the possible possible_error values or do some type of lookup

Reply:
yeah, that's straightforward! i wasn't sure which way was less brittle 🤷‍♀️ updated here: https://github.com/18F/identity-idp/pull/12060/commits/1f63c2bac9d6af40a9c42bb1ef27fd991feee491